llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -502,6 +502,11 @@ class TargetLoweringBase {
     return HasMultipleConditionRegisters;
   }
 
+  /// Return true if multiple vector predicate registers are available.
+  bool hasMultipleVectorPredicateRegisters() const {
+    return HasMultipleVectorPredicateRegisters;
+  }
```

Comment:
Imo this should just take a type and be `hasMultiplePredicateRegister(Type * Ty)`